### PR TITLE
feat: Add profile management screen

### LINKED ISSUE #14: Add profile management screen
We need a profile management screen in this app.

I will be documenting how this works in this document: https://github.com/ory/docs/blob/kratos/docs/kratos/selfservice/flows/profile.md

### LINKED ISSUE #24: Migrate to new user-facing error API
See https://github.com/ory/kratos/pull/219 for reference

### PR DESCRIPTION
**Pull request by zepatrik**
closes #14 

as the error screen is very primitive just changing the url to fetch errors from already closes #24, as I have done in this pr as well

### CODE REVIEW

File: src/helpers.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+class ErrorWithRedirect extends Error {
+  redirectTo: string
+
+  constructor(redirectTo: string) {
+    super(`There was an error that is solvable by redirecting to ${redirectTo}.`);
+    this.redirectTo = redirectTo
+  }
+}
+
+export const withResource = ({
```
Comment (aeneasr) [inline src/helpers.ts]: I think this is a pretty good idea (writing a wrapper around the fetching part). I've just checked out the SDK we generate for typescript, and I think it's pretty good and easy to use:

```
import kratos from '@oryd/kratos-client'
import {HttpError} from "@oryd/kratos-client/api/apis";

const api = new kratos.AdminApi('http://path/to/kratos')
api.getSelfServiceBrowserLoginRequest('asdf')
    .then(({response, body}) => {
    // body.methods.password.config.action
})
    .catch((err) => {
        if (err instanceof HttpError) {
            // err.statusCode == 404
        }
    })
```

What do you think about pausing this PR, moving to the SDK for auth endpoints and once that is merged, use the same methodology for the profile management?

Benefits would include:

- Future updates to paths just need a bump in the SDK
- Payloads are already pre-defined (we can delete the types.ts file)
- Easier to use and understand for developers

Comment (zepatrik) [reply src/helpers.ts]: Good idea

File: typings/types/index.d.ts [hunk 2]
```diff
@@ -13,7 +13,19 @@ declare module 'types' {
 
     export type Errors = Object[]
 
-    export interface Config {
+    export interface ProfileConfig {
```
Comment (aeneasr) [inline typings/types/index.d.ts]: Is this still required, now that we're using the TS code?

Comment (zepatrik) [reply typings/types/index.d.ts]: right no

**Pull request merged**